core-rust-bridge/src/main/java/com/radixdlt/protocol/ProtocolConfig.java

```diff
@@ -105,6 +119,43 @@ public static ProtocolConfig onlyGenesis() {
     return new ProtocolConfig(ImmutableList.of());
   }
 
+  public static ProtocolConfig launchAt(String upToProtocolVersionName) {
+    String newVersion = null;
+    List<ProtocolUpdateTrigger> protocolUpdateTriggers = new ArrayList<>();
+    for (String version : VERSION_NAMES) {
+      var lastVersion = newVersion;
+      newVersion = version;
+      if (lastVersion != null) {
+        protocolUpdateTriggers.add(
+            new ProtocolUpdateTrigger(newVersion, immediatelyAfter(lastVersion)));
+      }
+      if (newVersion.equals(upToProtocolVersionName)) {
+        break;
+      }
+    }
+    return new ProtocolConfig(ImmutableList.copyOf(protocolUpdateTriggers));
+  }
+
+  public static ProtocolConfig enactAtEpoch(String upToProtocolVersionName, long epoch) {
+    String newVersion = null;
+    List<ProtocolUpdateTrigger> protocolUpdateTriggers = new ArrayList<>();
+    for (String version : VERSION_NAMES) {
+      var lastVersion = newVersion;
+      newVersion = version;
+      if (lastVersion != null) {
+        if (newVersion.equals(upToProtocolVersionName)) {
+          protocolUpdateTriggers.add(
+              new ProtocolUpdateTrigger(newVersion, unconditionallyAtEpoch(epoch)));
+          break;
+        } else {
+          protocolUpdateTriggers.add(
+              new ProtocolUpdateTrigger(newVersion, immediatelyAfter(lastVersion)));
+        }
+      }
+    }
```

Comment:
Slightly confused by the logic, shouldn't we have
```
Genesis
P1 -> unconditional at (epoch)
P2 -> immediately after (P1)
P3 -> immediately after (P2)
```

Reply:
This takes a protocol version. It's used to have, say, Anemone / Bottlenose enact at genesis, then do some tests, then enact Cuttlefish a bit later. I've added a comment to explain :)